test/e2e/fixtures/DataGrid/ConcurrentReactUpdate.tsx

```diff
@@ -0,0 +1,15 @@
+import * as React from 'react';
+import { DataGrid, GridRowModel } from '@material-ui/data-grid';
+
+const columns = [{ field: 'id', headerName: 'Id', width: 100 }];
+
+export default function ConcurrentReactUpdate() {
```

Comment:
is this test not working on my PR?

Reply:
When I tried to run https://github.com/mui-org/material-ui-x/pull/1807/files#diff-4b13b5b8ec20f6d20174b3716e9e03501d374fe0ca6b92c5faeb5ea20f3282b2R396 on master, it was passing.

Reply:
Note that this test only runs on ChromeHeadless, not on Safari like in #1807, so definitely a tradeoff.

Reply:
then why don't we put it in my PR?

Reply:
We can, for sure. The idea was to demonstrate a better test, and a fix closer to the root cause.